language/move-binary-format/src/file_format.rs

```diff
@@ -249,8 +249,38 @@ pub struct StructHandle {
     /// For any instantiation of this type, the abilities of this type are predicated on
     /// that ability being satisfied for all type parameters.
     pub abilities: AbilitySet,
-    /// The type formals (identified by their index into the vec) and their constraints
-    pub type_parameters: Vec<AbilitySet>,
+    /// The type formals (identified by their index into the vec)
+    pub type_parameters: Vec<TypeParameter>,
+}
+
+/// A type parameter used in the declaration of a struct.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+pub struct TypeParameter {
```

Comment:
nit, maybe name this `StructTypeParameter` as it isn't used by functions. Or some other name to indicate that

Reply:
Yeah, I was thinking the same.